test/lsp.py

```diff
@@ -1445,8 +1452,11 @@ def test_generic(self, solc: JsonRpcProcess) -> None:
 
         for sub_dir in map(lambda filepath: filepath.name, sub_dirs):
             tests = map(
-                lambda filename, sd=sub_dir: sd + "/" + filename[:-len(".sol")],
-                os.listdir(f"{self.project_root_dir}/{sub_dir}")
+                lambda file_object, sd=sub_dir: sd + "/" + file_object.name[:-len(".sol")],
+                filter(
+                    lambda filepath: filepath.is_file() and filepath.name.endswith('.sol'),
+                    os.scandir(f"{self.project_root_dir}/{sub_dir}")
+                )
             )
```

Comment:
I never really warmed up to them because the syntax is so unintuitive :sweat_smile:

Reply:
this code example isn't necessarily more readable, IMHO. To be fair, using filter()/map() isn't either. I was simply fixing a problem here. :-)

Reply:
But yeah. I know it's not the point of the PR. I approved it anyway. Just saying that this part could be simplified.

Reply:
> I never really warmed up to them because the syntax is so unintuitive

I think it's a matter of getting used to them. Honestly, for me the `map()`/`filter()` version is something I have to translate in my head into plain code with comprehensions :)